lib/mayaUsd/fileio/primUpdaterManager.cpp

```diff
@@ -132,6 +132,36 @@ SdfPath makeDstPath(const SdfPath& dstRootParentPath, const SdfPath& srcPath)
     return dstRootParentPath.AppendPath(relativeSrcPath);
 }
 
+//------------------------------------------------------------------------------
+//
+// Verify if the given prim under the given UFE path is already edited or
+// is an ancestor of an already edited prim.
+bool isAlreadyEditedAsMaya(const Ufe::Path& ufePulledPath)
+{
+    MObject pullSetObj;
+    auto    status = UsdMayaUtil::GetMObjectByName(kPullSetName, pullSetObj);
+    if (status != MStatus::kSuccess)
+        return false;
+
+    MFnSet         fnPullSet(pullSetObj);
+    MSelectionList members;
+    const bool     flatten = true;
+    fnPullSet.getMembers(members, flatten);
+
+    for (unsigned int i = 0; i < members.length(); ++i) {
+        MDagPath itemPath;
+        members.getDagPath(i, itemPath);
+        Ufe::Path itemUfePath;
+        if (!PrimUpdaterManager::readPullInformation(itemPath, itemUfePath))
+            continue;
+
+        if (ufePulledPath.startsWith(itemUfePath) || itemUfePath.startsWith(ufePulledPath))
```

Comment:
I'm confused about this.  How can a user edit as Maya a descendant of something that has already been pulled?  By definition, all descendants of a pulled prim are also pulled.  Therefore:

if (ufeQueryPath.startsWith(pulledUfePath) --> this means an already pulled UFE path is an ancestor of the query path.  Therefore, the query path is already edited as Maya.  This should never happen, and should be a TF_VERIFY, no?

|| pulledUfePath.startsWith(ufeQueryPath) --> this means the query path is an ancestor of a pulled path.  This does NOT mean the query path is already edited as Maya, which is what the function name says.  It just means the query argument is an ancestor of an already pulled path, and we want to prevent this query path from being edited.

I think the function should be renamed to hasEditedDescendant(), and we should assert that the query path is not a descendant of an already-pulled prim, which would in my opinion be demonstrating a programming error.

Reply:
I wrote the function to be fool-proof without dependencies on what is possible now vs what could be possible in the future. I almost called it something like "wouldCauseDoubleEditions()" but I could not find a non-awkward name.

I can limit it to what is currently possible and follow your suggestions.